### PR TITLE
Cache SCSS dependency file in memcache

### PR DESCRIPTION
**Pull request by rullzer**
Since reading a file from disks can be costly. Lets store the dependency
json also in memcache.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by LukasReschke — DISMISSED** [review]
Tests don't approve this 😉

**Review by MorrisJobke — DISMISSED** [review]
I forgot to submit my feedback 🙈

File: lib/private/TemplateLayout.php [hunk 1]
```diff
@@ -218,7 +218,8 @@ static public function findStylesheetFiles($styles, $compileScss = true) {
 				\OC::$server->getAppDataDir('css'),
 				\OC::$server->getURLGenerator(),
 				\OC::$server->getConfig(),
-				\OC::$SERVERROOT
+				\OC::$SERVERROOT,
+				\OC::$server->getMemCacheFactory()->create('SCSS')
```
Comment (rullzer) [inline lib/private/TemplateLayout.php]: O joy. Ok let me create a local one then ;)

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Review by LukasReschke — APPROVED** [review]

File: lib/private/TemplateLayout.php [hunk 2]
```diff
@@ -213,12 +213,15 @@ static public function findStylesheetFiles($styles, $compileScss = true) {
...
+			/** @var \OC\Memcache\Factory $cache */
+			$cache = \OC::$server->query('MemCacheFactory');
 			$SCSSCacher = new SCSSCacher(
 				\OC::$server->getLogger(),
 				\OC::$server->getAppDataDir('css'),
 				\OC::$server->getURLGenerator(),
 				\OC::$server->getConfig(),
-				\OC::$SERVERROOT
+				\OC::$SERVERROOT,
+				$cache->createLocal('SCSS')
```
Comment (MorrisJobke) [inline lib/private/TemplateLayout.php]: With #4798  we should use the normal distributed cache as well.

File: lib/private/TemplateLayout.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/TemplateLayout.php]: See #4798 - we now store this properly in the object storage as well ;)

**Pull request merged**